File: src/app/interfaces/event/WebsiteButtons.java
package app.interfaces.event;

import app.interfaces.dialog.ErrorMessage;

import java.awt.*;
import java.awt.event.ActionEvent;
import java.net.URI;
/**
 * @author Jack Meng
 */
public class WebsiteButtons implements java.awt.event.ActionListener {
  private final String url;

  public WebsiteButtons(String url) {
    this.url = url;
  }

  @Override
  public void actionPerformed(ActionEvent e) {
    try {
      Desktop.getDesktop().browse(new URI(url));
    } catch (java.io.IOException | java.net.URISyntaxException e1) {
      e1.printStackTrace();
      new ErrorMessage(java.util.Arrays.toString(e1.getStackTrace()));
    }
    
  }
  
}

File: kt/app/interfaces/event/WebsiteButtons.kt
package app.interfaces.event

import app.interfaces.dialog.ErrorMessage
import java.awt.Desktop
import java.awt.event.ActionEvent
import java.awt.event.ActionListener
import java.io.IOException
import java.net.URI
import java.net.URISyntaxException
import java.util.*

/**
 * @author Jack Meng
 */
class WebsiteButtons(private val url: String) : ActionListener {
    override fun actionPerformed(e: ActionEvent) {
        try {
            Desktop.getDesktop().browse(URI(url))
        } catch (e1: IOException) {
            e1.printStackTrace()
            ErrorMessage(Arrays.toString(e1.stackTrace))
        } catch (e1: URISyntaxException) {
            e1.printStackTrace()
            ErrorMessage(Arrays.toString(e1.stackTrace))
        }
    }
}